agent/lib_monolog.c

```diff
@@ -397,21 +544,30 @@ NR_PHP_WRAPPER(nr_monolog_logger_addrecord) {
   int api = 0;
   size_t argc = 0;
   char* message = NULL;
+  nr_attributes_t* context_attributes = NULL;
   nrtime_t timestamp = nr_get_time();
 
   /* Values of $message and $timestamp arguments are needed only if log
    * forwarding is enabled so agent will get them conditionally */
   if (nr_txn_log_forwarding_enabled(NRPRG(txn))) {
     argc = nr_php_get_user_func_arg_count(NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
     message = nr_monolog_build_message(argc, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+
+    if (nr_txn_log_forwarding_context_data_enabled(NRPRG(txn))) {
+      zval* context_data = nr_monolog_extract_context_data(
+          argc, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
+      context_attributes
+          = nr_monolog_convert_context_data_to_attributes(context_data);
+      nr_php_arg_release(&context_data);
+    }
     api = nr_monolog_version(this_var TSRMLS_CC);
     timestamp
         = nr_monolog_get_timestamp(api, argc, NR_EXECUTE_ORIG_ARGS TSRMLS_CC);
   }
 
   /* Record the log event */
   nr_txn_record_log_event(NRPRG(txn), level_name, message, timestamp,
-                          NRPRG(app));
+                          context_attributes, NRPRG(app));
```

Comment:
Is the ownership of `context_attributes` transferred to `NRPRG(txn)` and `context_attributes` will get released when `NRPRG(txn)` is destroyed?

Reply:
I see it happening in `nr_log_event_destroy`.